docs/configure-rails/yaml-schema/configuration-reference.md

```diff
@@ -0,0 +1,744 @@
+| `content safety check input` | NVIDIA content safety model |
+| `topic safety check input` | Topic control model |
+
+### Output Rails
+
+Process LLM responses before returning to users.
+
+```yaml
+rails:
+  output:
+    parallel: false
+    flows:
+      - self check output
+      - self check facts
+    streaming:
+      enabled: false
+      chunk_size: 200
+      context_size: 50
+      stream_first: true
+```
+
+| Attribute | Type | Default | Description |
+|-----------|------|---------|-------------|
+| `parallel` | boolean | `false` | Execute output rails in parallel |
+| `flows` | list | `[]` | Names of flows that implement output rails |
+| `streaming` | object | | Streaming output configuration |
+
+#### Output Streaming Configuration
+
+| Attribute | Type | Default | Description |
+|-----------|------|---------|-------------|
+| `enabled` | boolean | `false` | Enable streaming mode |
+| `chunk_size` | integer | `200` | Tokens per processing chunk |
+| `context_size` | integer | `50` | Tokens carried from previous chunk |
+| `stream_first` | boolean | `true` | Stream before applying output rails |
+
+#### Built-in Output Flows
+
+| Flow | Description |
+|------|-------------|
+| `self check output` | LLM-based policy compliance check |
+| `self check facts` | Fact verification |
+| `self check hallucination` | Hallucination detection |
+| `mask sensitive data on output` | Mask PII in output |
+| `llama guard check output` | LlamaGuard content moderation |
+| `content safety check output` | NVIDIA content safety model |
+
+### Retrieval Rails
+
+Process chunks retrieved from knowledge base.
+
+```yaml
+rails:
+  retrieval:
+    flows:
+      - check retrieval sensitive data
+```
+
+### Dialog Rails
+
+Control conversation flow after user intent is determined.
+
+```yaml
+rails:
+  dialog:
+    single_call:
+      enabled: false
+      fallback_to_multiple_calls: true
+    user_messages:
+      embeddings_only: false
+      embeddings_only_similarity_threshold: null
+      embeddings_only_fallback_intent: null
+```
+
+| Attribute | Type | Default | Description |
+|-----------|------|---------|-------------|
+| `single_call.enabled` | boolean | `false` | Use single LLM call for intent + response |
+| `single_call.fallback_to_multiple_calls` | boolean | `true` | Fall back if single call fails |
+| `user_messages.embeddings_only` | boolean | `false` | Use only embeddings for intent matching |
+
+### Action Rails
+
+Control custom action and tool invocations.
+
+```yaml
+rails:
+  actions:
+    instant_actions:
+      - action_name_1
+      - action_name_2
+```
+
+### Tool Rails
+
+Control tool input/output processing.
+
+```yaml
+rails:
+  tool_output:
+    flows:
+      - validate tool parameters
+    parallel: false
+
+  tool_input:
+    flows:
+      - filter tool results
+    parallel: false
+```
+
+---
+
+## Rails Config Section
+
+The `rails.config` section contains configuration for specific built-in rails.
+
```

Comment:
This section will need links to further details for each, especially the 3rd party rails. Can be done in a later PR.